x/sigs/handler_test.go

```diff
@@ -0,0 +1,131 @@
+package sigs
+
+import (
+	"context"
+	"testing"
+
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/orm"
+	"github.com/iov-one/weave/store"
+	"github.com/iov-one/weave/weavetest"
+)
+
+func TestBumpSequence(t *testing.T) {
+	var (
+		key1 = weavetest.NewKey().PublicKey()
+		key2 = weavetest.NewKey().PublicKey()
+	)
+
+	cases := map[string]struct {
+		// Before performing the test, initialize the database with given user data.
+		InitData       []*UserData
+		Msg            BumpSequenceMsg
+		WantCheckErr   *errors.Error
+		WantDeliverErr *errors.Error
+		WantSequences  []*UserData
+	}{
+		"great success": {
+			InitData: []*UserData{
+				{Pubkey: key1, Sequence: 1},
+				{Pubkey: key2, Sequence: 9},
+			},
+			Msg: BumpSequenceMsg{
+				Pubkey:    key1,
+				Increment: 1,
+			},
+			WantSequences: []*UserData{
+				{Pubkey: key1, Sequence: 2},
+				{Pubkey: key2, Sequence: 9},
+			},
+		},
+		"message with a missing public key is invalid": {
+			Msg:          BumpSequenceMsg{Pubkey: nil, Increment: 1},
+			WantCheckErr: errors.ErrInvalidMsg,
+		},
+		"message with a negative sequence increment is invalid": {
+			InitData: []*UserData{
+				{Pubkey: key1, Sequence: 1},
+			},
+			Msg:          BumpSequenceMsg{Pubkey: key1, Increment: -1},
+			WantCheckErr: errors.ErrInvalidMsg,
+		},
+		"message with a zero sequence increment is invalid": {
+			InitData: []*UserData{
+				{Pubkey: key1, Sequence: 1},
+			},
+			Msg:          BumpSequenceMsg{Pubkey: key1, Increment: 0},
+			WantCheckErr: errors.ErrInvalidMsg,
+		},
+		"user that we increment the sequence of must exist": {
+			InitData:     nil,
+			Msg:          BumpSequenceMsg{Pubkey: key1, Increment: 421},
+			WantCheckErr: errors.ErrNotFound,
+		},
+		"sequence increment value must not be greater than 1000": {
+			InitData: []*UserData{
+				{Pubkey: key1, Sequence: 4},
+			},
+			Msg:          BumpSequenceMsg{Pubkey: key1, Increment: 1001},
+			WantCheckErr: errors.ErrInvalidMsg,
+		},
+		"sequence increment value can be 1000": {
+			InitData: []*UserData{
+				{Pubkey: key1, Sequence: 4},
+			},
+			Msg: BumpSequenceMsg{Pubkey: key1, Increment: 1000},
+			WantSequences: []*UserData{
+				{Pubkey: key1, Sequence: 1004},
+			},
```

Comment:
req: please also check:
- [ ] we do not overflow: current seq MaxInt64 -2 + bump 2 ->must be rejected (as it is +1 for the Tx already)
- [ ] pubKey owner has signed the msg (or drop that field in favour of main signer)

Reply:
pp: testing the increment value `1` would be good as well as it is the lower boundary and needs to work